pkg/generators/rules/idl_tag_test.go

```diff
@@ -0,0 +1,65 @@
+package rules
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/gengo/types"
+)
+
+func TestListTypeMissing(t *testing.T) {
+	tcs := []struct {
+		// name of test case
+		name string
+		t    *types.Type
+
+		// expected list of violation fields
+		expected []string
+	}{
+		{
+			name:     "none",
+			t:        &types.Type{},
+			expected: []string{},
+		},
+		{
+			name: "simple missing",
+			t: &types.Type{
+				Kind: types.Struct,
+				Members: []types.Member{
+					types.Member{
+						Name: "Containers",
+						Type: &types.Type{
+							Kind: types.Slice,
+						},
+					},
+				},
+			},
+			expected: []string{"Containers"},
+		},
+		{
+			name: "simple passing",
+			t: &types.Type{
+				Kind: types.Struct,
+				Members: []types.Member{
+					types.Member{
+						Name: "Containers",
+						Type: &types.Type{
+							Kind: types.Slice,
+						},
+						CommentLines: []string{"+listType=map"},
```

Comment:
@apelisse is this something we need in CRDs via a `x-kubernetes-*` extension?

Reply:
Yes, and a few others, let's talk